Group-writeable directories in the hierarchy above where we
run the tests from can cause failures due to openssh's strict
permissions checks. Adding this option helps the tests to run
more reliably on a variety of build systems.

--- tests/test_sshd.test
+++ tests/test_sshd.test
@@ -71,6 +71,7 @@ chmod go-rwx \
 # shellcheck disable=SC2086
 "${SSHD}" \
   -f "${SSHD_FIXTURE_CONFIG:-${d}/openssh_server/sshd_config}" \
+  -o 'StrictModes no' \
   -o 'Port 4711' \
   -h "${d}/openssh_server/ssh_host_rsa_key" \
   -h "${d}/openssh_server/ssh_host_ecdsa_key" \
